python/test/unit/io/test_adios2.py

```diff
@@ -312,19 +312,17 @@ def test_vtx_reuse_mesh(self, tempdir, dim, simplex, reuse):
 
         reuse_variables = ["NumberOfEntities", "NumberOfNodes", "connectivity", "geometry", "types"]
         target_all = 3  # For all other variables the step count is number of writes
-        target_mesh = (
-            1 if reuse else 3
-        )  # For mesh variables the step count is 1 if reuse else number of writes
-
-        # backwards compatibility adios2 < 2.10.0
-        try:
-            adios_file = adios2.open(str(filename), "r", comm=mesh.comm, engine_type="BP4")
-        except AttributeError:
-            # adios2 >= v2.10.0
+        target_mesh = 1 if reuse else 3
+        # For mesh variables the step count is 1 if reuse else number of writes
+        if hasattr(adios2, "is_built_with_mpi"):
```

Comment:
This is overcomplicated. Looking at the ADIOS2 code, the attribute `is_built_with_mpi` is always defined. If it's not in earlier versions, put ADIOS2 version check in a marker for the whole test.

Reply:
Version checks are not trivial without external libraries (i think), Ref https://til.simonwillison.net/python/comparing-version-numbers

Reply:
True.  I wasn't sure how easy it would be to set up which is why I didn't mention it.
mpi4py uses [versioned conditions](https://github.com/mpi4py/mpi4py/blob/fd1abfeb86305623e4381801f3a99b70f6505e51/test/test_spawn.py#L86) but that's implemented via [custom rules](https://github.com/mpi4py/mpi4py/blob/fd1abfeb86305623e4381801f3a99b70f6505e51/test/mpiunittest.py#L27). Similar h5py has [version conditions](https://github.com/h5py/h5py/blob/862eecefc72eb6941db3b370dff27a6e18bf6197/h5py/tests/test_file.py#L238), but on its own version provided as a tuple.

Reply:
That said, it could be simplified with 
```
@pytest.mark.skipif(not hasattr(adios2, "is_built_with_mpi"), "requires adios2>=2.10.0")
```

Reply:
Thats not quite right as adios2 is only defined through importorskip

Reply:
importorskip would already skip the entire test file though.
Then too, importorskip has its own [version parsing](https://docs.pytest.org/en/6.2.x/skipping.html#skipping-on-a-missing-import-dependency)
```
adios2 = pytest.importorskip("adios2", minversion="2.10.0")
```